pkg/metrics/ensuarance.go

```diff
@@ -0,0 +1,231 @@
+package metrics
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	k8smetrics "k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+// This const block defines the metric names for the crane-agent metrics.
+const (
+	CraneNamespace      = "crane"
+	CraneAgentSubsystem = "craneAgent"
+
+	LastActivity                = "last_activity"
+	StepDurationSeconds         = "step_duration_seconds"
+	StepDurationQuantileSummary = "step_duration_quantile_summary"
+
+	AnalyzerStatus      = "analyzer_status"
+	AnalyzerStatusTotal = "analyzer_status_total"
+	ExecutorStatus      = "executor_status"
+	ExecutorStatusTotal = "executor_status_total"
+	ExecutorErrorTotal  = "executor_error_total"
+	ExecutorEvictTotal  = "executor_evict_total"
+)
+
+type StepLabel string
+
+const (
+	StepMain         StepLabel = "main"
+	StepCollect      StepLabel = "collect"
+	StepAvoid        StepLabel = "avoid"
+	StepRestore      StepLabel = "restore"
+	StepUpdateConfig StepLabel = "updateConfig"
+)
+
+type SubComponent string
+
+const (
+	SubComponentSchedule SubComponent = "schedule"
+	SubComponentThrottle SubComponent = "throttle"
+	SubComponentEvict    SubComponent = "evict"
+)
+
+type AnalyzeType string
+
+const (
+	AnalyzeTypeEnableScheduling AnalyzeType = "enableScheduling"
+	AnalyzeTypeAvoidance        AnalyzeType = "avoidance"
+	AnalyzeTypeRestore          AnalyzeType = "restore"
+	AnalyzeTypeAnalyzeError     AnalyzeType = "analyzeError "
+)
+
+const (
+	// LogLongDurationThreshold defines the duration after which long step
+	// duration will be logged (in addition to being counted in metric).
+	// This is meant to help find unexpectedly long step execution times for
+	// debugging purposes.
+	LogLongDurationThreshold = 1 * time.Minute
+)
+
+var (
+	/**** Metrics related to crane-agent execution ****/
+	lastActivity = k8smetrics.NewGaugeVec(
+		&k8smetrics.GaugeOpts{
+			Namespace:      CraneNamespace,
+			Subsystem:      CraneAgentSubsystem,
+			Name:           LastActivity,
+			Help:           "Last time certain part of crane logic executed.",
```

Comment:
what logic? can you make it more specific?

Reply:
done~